### DISCUSSION TITLE
How to use power features

### DISCUSSION THREAD

**Discussion by adlerweb** [netbox-community/netbox#10406] [Q&A] [answered]
Morning,

I'm currently stuggeling to find enough information on how the power features can be used. Maybe someone can shed some light onto what I am missing or where to find more documentation.

As per https://docs.netbox.dev/en/stable/features/power-tracking/ I understand I need a power panel, power feed and can chain devices using power ports (input) and power outlets (output) afterwards. While per #3377 power will not be calculated recursively it should at least be displayed per device.

For my setup: I use Netbox in Docker since somewhere around 2018 and am trying to keep it up to date, so running 3.3.4 at the moment. I already modeled some devices as PDUs in the past and got other devices/servers attached. I used this for cable/port tracking, but had no power draw configured so far. As new systems are to be deployed I tried to add power draw calculation to get an overview where the most reserves are available.

**Problem a: How to feed**

I created a power panel and power feed (AC, 240V, 32A, tried 3-Phase or 1-Phase). My PDUs are modeled based on a generic device and got a power port to source power. I can attach cables and choose other devices/outlets or feeds. While connecting to power outlets works as expected, feeds do not. I get the usual A/B-screen for cables and can search for the feed, however while found it is greyed out and the cursor resembles a 🚫. No matter what I choose as connector type, max/allocated power, etc. Sadly also no indication on why I am not allowed to connect there.

![grafik](https://user-images.githubusercontent.com/1146482/191052488-d19624a2-13a0-4424-9070-7d2e81620ac8.png)

**Problem b: No power?**

Even without a feed, calculation should still work, right? I set a maximum and allocated power draw on power ports of some devices and connected them to power outlets of my PDUs. 

![grafik](https://user-images.githubusercontent.com/1146482/191050368-b0e49371-97f5-4e14-a444-2c885fc1b19e.png)

The outlet port on the PDU is linked to a power port. While the outlet is shown in the PDUs device overview the power utilization remains zero

![grafik](https://user-images.githubusercontent.com/1146482/191051061-2d6b99b2-3bdb-419f-9d13-759a1e90b328.png)

I would have expected for the overview to reflect the 75W allocated draw of the connected device.

Any ideas what I am missing?

**Comment by cyberndj** [comment, marked as answer]
Howdy,

For problem A, I would suggest doing the connection from the power port section of the device (PDU) to the power feed. 

For problem B, what your showing is the PDU itself is going to use 1100VA. Now that would be some PDU... check out the note in this doc section. IE, PDU's should not have a power allocation associated with them.
https://docs.netbox.dev/en/stable/models/dcim/powerport/#maximum-draw

Side note...I think its a deficiency to not set some type of max power utilization and current utilization on a device modeled after a PDU.

**Reply by adlerweb** [reply]
a) Doh. Yeah, looking there helped. "Mark as connectd" was enabled on the feed.

b) OK, I thought max/allocated power would set the 100%-Mark (aka how much power is available for attached devices). I didn't expect to find more power documentation under DCIM. I guess this means my views will be stuck at 0% due to missing recursion for power circuits, as neither loads trickle down to the feed and the feed doesn't affect the PDUs.

At least I can now model everything in NetBox and calculate using external scripts.
